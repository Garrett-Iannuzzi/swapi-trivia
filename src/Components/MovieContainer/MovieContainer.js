import React from 'react';
import './MovieContainer.scss';
import MovieCard from '../MovieCard/MovieCard.js';
import PropTypes from 'prop-types';
import images from '../.././images/images.js';

const MovieContainer = ({ movieCards, getCharacterData }) => {
  const movies = movieCards.sort((a, b) => a.episode_id - b.episode_id).map(card => {
  const moviePosters = Object.keys(images.posters).find(posterTitle => {
    return posterTitle === card.title
  })
    return(
      <MovieCard 
        title={card.title}
        episode_id={card.episode_id}
        release_date={card.release_date}
        key={card.episode_id}
        getCharacterData={getCharacterData}
        poster={images.posters[moviePosters]}
      />
    );
  });
  return(
    <section>
      {movies}
    </section>
  );
};

export default MovieContainer;

MovieContainer.propTypes = {
  movieCards: PropTypes.arrayOf(PropTypes.object),
}
