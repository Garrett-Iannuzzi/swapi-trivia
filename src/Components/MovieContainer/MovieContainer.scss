section {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 84vh;
  justify-content: space-around;
  padding: 10px;
  overflow: scroll;
  width: 95%;
}
