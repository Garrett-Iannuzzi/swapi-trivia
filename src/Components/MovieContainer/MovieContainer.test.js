import React from 'react';
import { shallow } from 'enzyme';
import MovieContainer from '../MovieContainer/MovieContainer.js';

describe('MovieContainer', () => {

  it('Should match the snapshot with data passed in', () => {
    const movieMock = [
      {
        title: 'A New Hope',
        episode_id: 4,
        release_date: '1977-05-25',
      },
      {
        title: 'Empire Strikes Back',
        episode_id: 5,
        release_date:'1980-05-17' 
      }
    ]
    let wrapper = shallow(
      <section>
        <MovieContainer
          {...movieMock}
          getCharacterData={jest.fn()}
        />
      </section>
    );
    expect(wrapper).toMatchSnapshot();
  })
})