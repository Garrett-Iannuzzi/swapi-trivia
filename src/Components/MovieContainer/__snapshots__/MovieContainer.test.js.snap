// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer Should match the snapshot with data passed in 1`] = `
<section>
  <MovieContainer
    0={
      Object {
        "episode_id": 4,
        "release_date": "1977-05-25",
        "title": "A New Hope",
      }
    }
    1={
      Object {
        "episode_id": 5,
        "release_date": "1980-05-17",
        "title": "Empire Strikes Back",
      }
    }
    getCharacterData={[MockFunction]}
  />
</section>
`;
